core/deployment/src/main/java/io/quarkus/deployment/steps/SubstrateConfigBuildStep.java

```diff
@@ -104,6 +105,31 @@ void build(SslContextConfigurationTemplate sslContextConfigurationTemplate,
                 sslTrustStoreSystemProperty.produce(
                         new SslTrustStoreSystemPropertyBuildItem(
                                 graalVmLibDirectory.resolve(Paths.get("security", "cacerts")).toString()));
+
+                boolean enableJni = false;
+                for (JniBuildItem jniBuildItem : jniBuildItems) {
+                    enableJni = true;
+                    if (jniBuildItem.get().isPresent()) {
+                        Path libraryPath = Paths.get(jniBuildItem.get().get());
+                        if (!libraryPath.isAbsolute()) {
+                            try {
+                                Path absoluteLibPath = libraryPath.toAbsolutePath();
+                                log.warn("JNI library path is not absolute. " + libraryPath + " will be transformed as "
+                                        + absoluteLibPath);
+                                javaLibraryPathAdditionalPath
+                                        .produce(new JavaLibraryPathAdditionalPathBuildItem(absoluteLibPath.toString()));
+                            } catch (SecurityException e) {
+                                log.warn("JNI library path cannot be converted as absolute: " + libraryPath);
```

Comment:
Wondering if it should throw an exception instead?

If we decide to throw an exception, we could simplify this code a bit and produce the build item outside of the if.